### ISSUE TITLE
a way for world and or udon to be able to grab/get osc input

### ISSUE THREAD

**Issue by nagaisa** [vrchat-community/osc#110]
*What's the idea?**
a world would be able to get osc data and put it to use in ways with udon or other stuff so stuff like games and controlling stuff so let's say it can tell what you looking at and or if you blink. it can see heart rate and help tack burned cal or tell if you need a break

**Is there another way?**
not much for this part but mabey, just few parts can get it and use it and only if close/interacting with it

**Additional context**
it would be able to tell if it's from the master or player, the world should not be able to keep the data long (so prob around 1hr(or if sestion ends) after leaveing it get rid of it tho prob not right after they leave as the user may crash or need to rejoin and depending on whats used it might still  need it) also user should be able to set data deletion on there end as well so if they want it gone right after it will be or mabey for less time, as well as simply be able to turn on/off the sending to the world

**Comment by mjcox244** [comment]
This has a few fetures that can be a godsend for worlds.
1) if someone is making an avatar to accompany a world that uses OSC then the world can also tap into it in order to do things like send splatfest colours to people
2) worlds can use OSC in order to save progress in the world

There would be an output node that takes in a generic type and a sting with the OSC path
There would be an input node where you specify output type and a string with the path and it lets you read OSC data.

**Comment by Romaxx** [comment]
Yes, using osc player parameters (even readonly) could open more possibilities for worlds. Add it please :)

**Comment by EIA485** [comment]
> Yes, using osc player parameters (even readonly) could open more possibilities for worlds. Add it please :)

you can already do this sort of stuff with midi

**Comment by mjcox244** [comment]
> > you can already do this sort of stuff with midi

How? How do we hook this up?

**Comment by EIA485** [comment]
> > > you can already do this sort of stuff with midi
> 
> How? How do we hook this up?

[the docs](https://docs.vrchat.com/docs/midi)

**Comment by mjcox244** [comment]
> [the docs](https://docs.vrchat.com/docs/midi)

This is PC only and is input only
Tho this is a good excample of how it would work with OSC

OnOSCReceive block that has the string path it was sent to and the output (changeable with a drop-down)
With OSC you would also have a send block with yet again a path to send it and an object input
